scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -511,6 +531,47 @@ public void run() {
 			throw new HandshakeException("Cannot process handshake message", alert);
 		}
 	}
+	/**
+	 * Check, if message is expected.
+	 * 
+	 * @param message mesage to check
+	 * @throws HandshakeException if the message is not expected
+	 * @see #useStateValidation
+	 * @see DtlsConnectorConfig#useHandshakeStateValidation()
+	 */
+	protected void expectMessage(DTLSMessage message) throws HandshakeException {
+		if (useStateValidation && states != null) {
+			if (statesIndex >= states.length) {
+				LOGGER.warn("Cannot process {} message from peer [{}], no more expected!", HandshakeState.toString(message),
+						getSession().getPeer());
+				AlertMessage alert = new AlertMessage(AlertLevel.FATAL, AlertDescription.INTERNAL_ERROR,
+						session.getPeer());
+				throw new HandshakeException("Cannot process " + HandshakeState.toString(message)
+						+ " handshake message, no more expected!", alert);
+			}
+			HandshakeState expectedState = states[statesIndex];
+			boolean expected = expectedState.expect(message);
+			if (!expected && expectedState.isOptional()) {
+				if (statesIndex + 1 < states.length) {
+					HandshakeState nextExpectedState = states[statesIndex + 1];
+					if (nextExpectedState.expect(message)) {
+						++statesIndex;
+						expectedState = nextExpectedState;
```

Comment:
why setting expectedState ? (it seems used only if expected is false)

Reply:
OK